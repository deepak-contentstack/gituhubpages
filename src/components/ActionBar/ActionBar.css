@import '../../../tokens/variables.scss';
@import '../../../tokens/tokens.css';
@import '../../static/styles/css/typography.css';
@import '../../static/styles/css/common.css';

.hide {
  visibility: hidden;
}

.ActionBar {
  height: $space-20;
  padding: $space-10 0;
  &:hover {
    cursor: pointer;
    & > .hide {
      visibility: visible;
    }
  }
  &_bar {
    align-items: center;
    background-color: $color-brand-primary-base;
    display: flex;
    height: $space-2;
    justify-content: center;
    position: relative;
    &_disabled {
      background-color: $color-brand-secondary-base;
    }
  }
  &_sign {
    filter: drop-shadow(0 $space-4 $space-4 rgba(0, 0, 0, 0.25));
    height: $space-20;
    z-index: 100;
    .tippy-wrapper {
      height: $space-20;
    }
  }
}
