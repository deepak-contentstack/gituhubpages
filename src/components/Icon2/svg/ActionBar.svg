<svg width="12" height="12" viewBox="0 0 12 12" fill="none" xmlns="http://www.w3.org/2000/svg">
<g clip-path="url(#clip0)">
<path d="M8.5 9C9.32843 9 10 9.67157 10 10.5C10 11.3284 9.32843 12 8.5 12C7.67157 12 7 11.3284 7 10.5C7 9.67157 7.67157 9 8.5 9Z" fill="#475161"/>
<path d="M10 6C10 5.17157 9.32843 4.5 8.5 4.5C7.67157 4.5 7 5.17157 7 6C7 6.82843 7.67157 7.5 8.5 7.5C9.32843 7.5 10 6.82843 10 6Z" fill="#475161"/>
<path d="M8.5 0C9.32843 -3.62117e-08 10 0.671573 10 1.5C10 2.32843 9.32843 3 8.5 3C7.67157 3 7 2.32843 7 1.5C7 0.671573 7.67157 3.62117e-08 8.5 0Z" fill="#475161"/>
<path d="M3.5 9C4.32843 9 5 9.67157 5 10.5C5 11.3284 4.32843 12 3.5 12C2.67157 12 2 11.3284 2 10.5C2 9.67157 2.67157 9 3.5 9Z" fill="#475161"/>
<path d="M5 6C5 5.17157 4.32843 4.5 3.5 4.5C2.67157 4.5 2 5.17157 2 6C2 6.82843 2.67157 7.5 3.5 7.5C4.32843 7.5 5 6.82843 5 6Z" fill="#475161"/>
<path d="M3.5 0C4.32843 -3.62117e-08 5 0.671573 5 1.5C5 2.32843 4.32843 3 3.5 3C2.67157 3 2 2.32843 2 1.5C2 0.671573 2.67157 3.62117e-08 3.5 0Z" fill="#475161"/>
</g>
<defs>
<clipPath id="clip0">
<rect width="12" height="12" fill="white" transform="translate(0 12) rotate(-90)"/>
</clipPath>
</defs>
</svg>
