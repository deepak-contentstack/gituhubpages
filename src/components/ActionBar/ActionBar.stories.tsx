import { withA11y } from '@storybook/addon-a11y'
import { withKnobs } from '@storybook/addon-knobs'
import React from 'react'
import ActionBar from './ActionBar'
import notes from './ActionBar'


export default {
  title: 'Components / ActionBar',
  component: ActionBar,
  decorators: [withKnobs, withA11y],
  parameters: {
    componentSubtitle: 'Operation tool',
    notes: { notes },
  },
}

export const Default = () => {
  return (
    <ActionBar hoverMessage="Abc" onClick={() => console.log("Action bar clicked")} ><div>Abc</div></ActionBar>
  )
}

export const Disabled = () => {
  return (
    <ActionBar hoverMessage="Abc" onClick={() => console.log("Action bar clicked")} disabled={true} ><div>Abc</div></ActionBar>
  )
}