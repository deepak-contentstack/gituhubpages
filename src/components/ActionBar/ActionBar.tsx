import Tooltip from '../Tooltip/Tooltip';
import React from 'react';
// import Icon from '../Icon2';
import cn from 'classnames'
import './ActionBar.css'

export type ActionBarProps = {
  onClick: React.MouseEventHandler<HTMLDivElement>,
  hoverMessage: string | React.ReactNode | { text: string; icon: any }
  showOnHover?: boolean,
  children?: React.ReactChild,
  disabled?: boolean,
  testId?: string
}
const ActionBar = ({ children, onClick, hoverMessage, showOnHover = false, disabled = false, testId = "action-bar" }: ActionBarProps) => {
  const disableClickHandler = e => e.preventDefault()
  return (
    <>
      {React.Children.count(children) ? children : null}
      <div className="ActionBar" data-test-id={testId} onClick={(disabled ? disableClickHandler : onClick)}>
        <div className={cn('ActionBar_bar', { hide: showOnHover }, (disabled && "ActionBar_bar_disabled"))}>
          <div className="ActionBar_sign">
            <Tooltip content={hoverMessage} position="top"></Tooltip>
            This is test commit
          </div>
        </div>
      </div>
    </>
  );
};

export default ActionBar;