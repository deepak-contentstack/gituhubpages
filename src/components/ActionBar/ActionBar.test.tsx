import React from 'react'
import renderer from 'react-test-renderer'

import ActionBar from './ActionBar'

jest.mock('react-dom', () => {
  const original = jest.requireActual('react-dom');
  return {
    ...original,
    createPortal: node => node,
  };
});

it('renders the component', () => {
  const onClickFunction = jest.fn()
  const output = renderer.create(<ActionBar hoverMessage="Abc" onClick={onClickFunction} testId="action-bar">ActionBar</ActionBar>).toJSON();

  expect(output).toMatchSnapshot();
});