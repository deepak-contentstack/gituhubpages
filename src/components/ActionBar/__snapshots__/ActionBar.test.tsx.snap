// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
Array [
  "ActionBar",
  <div
    className="ActionBar"
    data-test-id="action-bar"
    onClick={[MockFunction]}
  >
    <div
      className="ActionBar_bar"
    >
      <div
        className="ActionBar_sign"
      >
        <div
          className="tippy-wrapper"
          data-test-id="cs-tooltip"
        />
        Abc
        This is test commit
      </div>
    </div>
  </div>,
]
`;
